common/test/views/support/cleaner/CommercialMPUForFrontsTest.scala

```diff
@@ -0,0 +1,167 @@
+                  <li class="fc-slice__item l-row__item"></li>
+                  <li class="fc-slice__item l-row__item"></li>
+                  <li class="fc-slice__item l-row__item fc-slice__item--mpu-candidate"></li>
+                </ul>
+              </div>
+            </div>
+          </section>
+          <section class="fc-container">
+            <div class="fc-container__inner">
+              <div class="fc-slice fc-slice--popular">
+              </div>
+            </div>
+          </section>
+          <div class="fc-container fc-container--commercial">
+            <div id="dfp-ad--merchandising" class="ad-slot--merchandising ad-slot--commercial-component">
+            </div>
+          </div>
+        </body>
+      </html>.toString()
+
+  it should "insert MPUs into applicable slices, and give them unique IDs" in {
+    val result = parseTestData(frontWithThrasher)
+    val body = clean(result)
+
+    val desktopMPUs = body.getElementsByClass("fc-slice__item--mpu-candidate")
+    desktopMPUs.size should be (3)
+    desktopMPUs.first.toString should include ("dfp-ad--inline1")
+    desktopMPUs.last.toString should include ("dfp-ad--inline3")
+  }
+
+  it should "insert MPUs for mobile view between sections, and give them unique IDs" in {
+    val result = parseTestData(frontWithThrasher)
+    val body = clean(result)
+
+    val mobileMPUs = body.getElementsByClass("fc-container__mpu--mobile")
+    mobileMPUs.size should be (2)
+    mobileMPUs.first.toString should include ("dfp-ad--top-above-nav--mobile")
+    mobileMPUs.last.toString should include ("dfp-ad--inline1--mobile")
+  }
+
+  it should "not count the first container, if it is a thrasher on a Network Front, when adding mobile MPUs" in {
+    val result = parseTestData(frontWithThrasher)
+    val body = clean(result)
+
+    val thrasher = body.getElementsByClass("fc-container--first").first
+    thrasher.id should be ("thrasher")
+
+    val sectionAfterThrasher = thrasher.nextElementSibling()
+    sectionAfterThrasher.hasClass("fc-container__mpu--mobile") should be (false)
+  }
+
+  it should "avoid inserting next to commercial containers when adding mobile MPUs" in {
+    val result = parseTestData(frontWithThrasher)
+    val body = clean(result)
+
+    val commercialContainers = body.getElementsByClass("fc-container--commercial")
+    commercialContainers.size should be (2)
+
+    commercialContainers.first.nextElementSibling.hasClass("fc-container__mpu--mobile") should be (false)
+    commercialContainers.first.previousElementSibling.hasClass("fc-container__mpu--mobile") should be (false)
+    commercialContainers.last.previousElementSibling.hasClass("fc-container__mpu--mobile") should be (false)
+
+    if (commercialContainers.last.nextElementSibling != null) {
+      commercialContainers.last.nextElementSibling.hasClass("fc-container__mpu--mobile") should be (false)
+    } else {
+      commercialContainers.last.nextElementSibling should be (null)
+    }
```

Comment:
Given the test data, is this always gonna end up executing the same `if` branch?